### ISSUE TITLE
solved laravel-passport problem but laravel-permission not yet

### ISSUE THREAD

**Issue by kerim92** [gecche/laravel-multidomain#51]
Hi,

We are creating a big crm / acaunting software that we wanted to prefer laravel-multidomain package for our project for run only one laravel installation with multi databases.

at the first install we have got an error with laravel-passport package, error was "Unable to read key" \vendor\league\oauth2-server\src\CryptKey.php:77

![key](https://user-images.githubusercontent.com/56412326/122111578-947a7880-ce28-11eb-834c-b4eab0b1e9cb.png)

We can not find any quick solve subject on internet. After few try, it was very easy to solve with following cli command
php artisan passport:keys --domain=sub.domainname.com

we are using spatie/laravel-permission package for for controlling user roles.
many functions our software are seems working properly (for now) expect one,

![passport](https://user-images.githubusercontent.com/56412326/122112976-259e1f00-ce2a-11eb-8153-143b96f7559a.png)

when we go to user edit function, we have got following error
BadMethodCallException Call to undefined method App\Models\User\User::getRoleIds() (View: /var/www/sub.domainname.com/html/resources/views/Panel/User/Edit.blade.php)

we are searching for a simple solution until find, i will be appricated if you suggest.

thanks in advance, 
Kerim

**Comment by gecche** [comment]
Hi,

I also use Spatie laravel-permission in nearly all my projects together with my Laravel-multidomain and I have no problem at all...

From your screenshots I don't think your problem is related to my package: it seems that your User model does not include Spatie trait... anyway I read from you screenshot "Call to undefined method App\Models\User\User::getRolesIds()": why App\Models\User\User and not only App\Models\User ?
I suspect that you have mistaken the class name of the User model.

Cheers 

Giacomo

**Comment by kerim92** [comment]
Hi Giacomo,

Yes you are right. i will close issue for now will update after solve.

Thank you!

**Comment by kerim92** [comment]
Hi Giacomo,

We have discovered problem. 

During vendor:publish i have selected all for publish but we have problems in our different vendors. So vendor:publish only your package solved problem. 

You are right we do not use laravel permission for our user roles. 

I have talked with our developers they wrote custom role system to our project. 

I have already given stars to your project 
We would like to support this project later as soon as possible. 

Thank you, 
Kerim
